src/headingengine.js

```diff
@@ -30,6 +23,39 @@ export default class HeadingEngine extends Plugin {
 	/**
 	 * @inheritDoc
 	 */
+	constructor( editor ) {
+		super( editor );
+
+		const t = editor.t;
+
+		/**
+		 * A set of default localized labels for `config.heading.formats`.
+		 *
+		 * @readonly
+		 * @protected
+		 * @member {Object} #_localizedFormatLabels
+		 */
+		const labels = this._localizedFormatLabels = {
```

Comment:
I'm thinking about the localization of user config and I'm afraid about one thing. Perhaps it's an edge case, but still...

Let's say you want editor in language X. So the `_localizedOptionLabels` contains e.g. map like:

```
Paragraph: 'X Paragraph',
'Heading 1': 'X Heading 1',
...
```

Now, someone also specifies `config.heading.options`, but want to use English strings there because for a weird case it make sense to keep them in English. But this turns to be impossible because they are automatically localized using `_localizedOptionLabels` mapping.

I hope that this case is really that edgy as it sounds now and can be ignored.

PS. This applies to the current implementation and the one proposed by me above.